### ISSUE TITLE
Tasks in chain lost task name

### ISSUE THREAD

**Issue by junnikokuki** [kodless/leek#51]
I have a chain task which has two tasks, the first task shows right in Tasks page, but the second shows "Task name not yet received" in task name column and no route and queue info. But the second task shows right in Flower. The two are in different queues.

**Comment by junnikokuki** [comment]
Found that it is a bug when api validate the payload. In chain tasks, the child task except will not generate the task-sent event. In that situation, in my environment, the workers will generate 3 events in one payload, it is task-received, task-started and succeed. But in leek/api/schemas/serializer.py' validate_payload() function, this will generate 3 Task with same id, so when validated_payload.update(validate_event(event, app_env)), Task will be updated with last succeed status which has no name.

**Comment by kodless** [comment]
@junnikokuki Leek performs upserts to update tasks, task name is sent only by clients' `task-sent` event and workers `task-started` event. if you didn't configure your clients to send `task-sent` events and you didn't configure your workers to send `task-started` events, leek will not be able to show the task names. follow celery docs to instruct celery clients to send `task-sent` events https://docs.celeryproject.org/en/stable/userguide/configuration.html#task-send-sent-event and instruct celery workers to send `task-started` events https://docs.celeryproject.org/en/stable/userguide/configuration.html#task-track-started

Leek is showing `Task name not yet received` because events are sent unordered and if `task-succeeded` event is received before `task-sent` event, then leek will show `Task name not yet received`. if your celery clients(eg: api) did not send `task-sent` event at all. leek will keep showing `Task name not yet received`.

`task-succeeded` event does not remove task name from task document because task name is not sent by `task-succeeded` event and Leek is upserting not overwriting task document, you can check celery docs https://docs.celeryproject.org/en/stable/userguide/monitoring.html#task-succeeded

Please follow Leek requirements guide to make sure you have the best monitoring experience https://tryleek.com/docs/introduction/requirements

**Comment by junnikokuki** [comment]
Yes, I have configured the client to send task-sent event, but the chain task which has child tasks will not generate the task-sent event of the non-first-child-task. Maybe this is a celery bug.

**Comment by junnikokuki** [comment]
Glad to see you have fix the bug in the recent commit. Thank you.

**Comment by kodless** [comment]
@junnikokuki  did your issue got resolved after upgrading to version 0.5.5? because I'm not sure if the recent refactoring in 0.5.5 somehow fixed the issue.

**Comment by junnikokuki** [comment]
Yes, the bug is fixed on my usage.

**Comment by kodless** [comment]
Awesome 🎉, closing!
